src/Microsoft.Identity.Web.Certificate/KeyVaultCertificateLoader.cs

```diff
@@ -46,11 +46,7 @@ public async Task LoadIfNeededAsync(CredentialDescription credentialDescription,
             X509KeyStorageFlags x509KeyStorageFlags)
         {
             Uri keyVaultUri = new Uri(keyVaultUrl);
-            DefaultAzureCredentialOptions options = new()
-            {
-                ManagedIdentityClientId = managedIdentityClientId,
-            };
-            DefaultAzureCredential credential = new(options);
+            ManagedIdentityCredential credential = new(managedIdentityClientId);

```

Comment:
This one is the KeyVault certificate loader.
Aren't you interested to test things locally on your box using dev credentials?

Reply:
I assumed that because a managedIdentityClientId is being used here unconditionally, it was only intended for MI scenarios.

Reply:
no. It's for all scenarios. This is the only way to test locally.

Reply:
I have resolved this.